internal/http/http.go

```diff
@@ -28,3 +32,33 @@ func SetCacheHeaders(w http.ResponseWriter, maxAge time.Duration, timeUntilExpir
 	w.Header().Set("Cache-Control", "public, max-age="+maxAge.String())
 	w.Header().Set("Expires", time.Now().Add(timeUntilExpiry).Format(time.RFC1123))
 }
+
+func LimitRead(r io.Reader, limit int64) ([]byte, error) {
+	lr := io.LimitReader(r, limit)
```

Comment:
I noticed we're not closing the request body anywhere else in the code, which we should be doing to prevent memory leaks. Since many handlers will all be reading request bodies using this, it's convenient if we close it right here. There's no harm if the caller _also_ happens to defer closing the request body.

This improvement might impact some unit tests, but you can wrap any and readers or buffers your tests use with a `io.NopCloser()`

```suggestion
func LimitRead(r io.ReaderCloser, limit int64) ([]byte, error) {
	defer r.Close()

	lr := io.LimitReader(r, limit)
```